File: backend/app/schemas/material_hypothesis.py
"""Material Hypothesis DTO"""
from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from typing import Any, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class MaterialHypothesisResponse(BaseModel):
    id: UUID
    scene_version_id: UUID
    target_type: str
    target_id: UUID
    material_name: str
    confidence: Optional[Decimal] = None
    source_method: Optional[str] = None
    is_selected: bool
    evidence_json: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime

File: frontend/src/types/material-hypothesis.ts
import type { ISODateString, UUID } from './common';

// 백엔드 §12.3 Material Hypothesis — 자동 추출된 재질 후보.
// scene_version_id 기반이므로 Draft 단계에선 비어있을 수 있음.

export interface MaterialHypothesis {
  id: UUID;
  scene_version_id: UUID;
  target_type: 'wall' | 'room' | 'opening' | 'object' | string;
  target_id: UUID;
  material_name: string;
  confidence: string | null;
  source_method: string | null;
  is_selected: boolean;
  evidence_json: Record<string, unknown>;
  created_at: ISODateString;
}
